src/createLanguageClient.ts

```diff
@@ -43,12 +43,22 @@ async function openConnection (url: URL | string, errorHandler: ConnectionErrorH
             if ((args[0] as MessageSignature).method === RegistrationRequest.type.method) {
               const registrationParams = params[0] as unknown as RegistrationParams
               registrationParams.registrations = registrationParams.registrations.filter(registration => {
-                return !existingRegistrations.has(registration.id)
+                const alreadyExisting = existingRegistrations.has(registration.id)
+                if (alreadyExisting) {
+                  console.warn('Registration already existing', registration.id, registration.method)
+                }
+                return !alreadyExisting
               })
               registrationParams.registrations.forEach(registration => {
                 existingRegistrations.add(registration.id)
               })
             }
+            if ((args[0] as MessageSignature).method === UnregistrationRequest.type.method) {
```

Comment:
nit: `args[0] as MessageSignature).method` could be extracted into a variable to simplify the 2 conditions.

Reply:
https://github.com/CodinGame/monaco-languageclient-wrapper/pull/4/commits/b2f61ab64de91e1f6e09dc483b61fc7834d04480